src/components/mc/cuda/mc_cuda.c

```diff
@@ -139,20 +142,23 @@ static ucc_status_t ucc_mc_cuda_mem_alloc(ucc_mc_buffer_header_t **h_ptr,
     h->from_pool = 0;
     h->mt        = UCC_MEMORY_TYPE_CUDA;
```

Comment:
shouldn't we use `mt` here ?

Reply:
no, there are certain cases (free) for example that we check what h->mt is and based on that choose what function to call, and besides alloc we want all other mc functions to treat cudaManages as cuda.